Add Cargo features for selecting installed CUDA version.

peterhj opened the pull request:
This commit adds feature flags in Cargo.toml for selecting a specific installed version of the CUDA toolkit. The design is basically the same as that used by clang-sys (https://github.com/KyleMayes/clang-sys/blob/master/Cargo.toml).

These feature flags are not currently used by the library or build script, but that could be done in another PR.

kngwyu commented:
What does `gte_` prefix mean?

And I think we should add a version flag after actually testing it.

AndrewGaspar commented:
> What does `gte_` prefix mean?

I assume it means "Greater-than-or-equal". I.e. the feature indicates at least that minimum version is available.

peterhj commented:
> What does `gte_` prefix mean?

The `gte_` (Greater Than or Equal) features can be used in the library itself, e.g. if certain APIs needed wrapping which differed between versions. The features _without_ `gte_` are specified in Cargo.toml by users of cuda-sys.

For example, one difference in CUDA APIs between versions is the addition of f16 support somewhere around 8.0 or 9.0, which added a new header among other changes.

peterhj commented:
I added some minimal usage of the new API version features (just for 8.0, since that's what cuda-sys binds to currently). I also updated the CI config to set the right features in the environment (more details in the top post).
